### PR TITLE
Update HAS for KCP

### PR DESCRIPTION
**Pull request by johnmcollier**
- Makes a number of changes to the kustomization manifests for deploying HAS on KCP
   - Separates the APIExport, APIResourceSchema, and APIBindings from the rest of the KCP manifests, to simplify deployment via infra-deployments
   - Adds a new `deploy-kcp` makefile target for deploying HAS on KCP
   - Updates the `docs/kcp.md` doc on deploying HAS on KCP
- Updates the KCP libraries in HAS for compatibility on CPS

### How to Test

1) Build the changes in this PR and push up to Quay
1) Create an OpenShift cluster (the workload cluster)
2) Follow the CPS onboarding document that outlines logging in kcp-stable, and setting up a synctarget / workload cluster
3) Follow the instructions in `docs/kcp.md`, and make sure ‘IMG` is set in your terminal when running `make deploy-kcp`

### CODE REVIEW

**Comment by maysunfaisal** [comment]
@johnmcollier seems like the tests are timing out on the condition checks

File: controllers/componentdetectionquery_controller.go [hunk 1]
```diff
@@ -67,7 +66,11 @@ type ComponentDetectionQueryReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("ComponentDetectionQuery", req.NamespacedName).WithValues("clusterName", req.ClusterName)
-	ctx = kcpclient.WithCluster(ctx, logicalcluster.New(req.ClusterName))
+
+	// if we're running on kcp, we need to include workspace in context
+	if req.ClusterName != "" {
+		ctx = logicalcluster.WithCluster(context.TODO(), logicalcluster.New(req.ClusterName))
```
Comment (maysunfaisal) [inline controllers/componentdetectionquery_controller.go:72]:
```suggestion
		ctx = logicalcluster.WithCluster(ctx, logicalcluster.New(req.ClusterName))
```
should be?

Comment (johnmcollier) [reply controllers/componentdetectionquery_controller.go:72]: Oof yes, that was some testing that was not supposed to be committed

File: Makefile [hunk 2]
```diff
@@ -176,12 +179,11 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
-deploy-kcp: manifests install ## Install CRDs and deploy HAS on KCP
+deploy-kcp: ensure-tmp manifests kustomize ##install ## Install CRDs and deploy HAS on KCP
```
Comment (maysunfaisal) [inline Makefile:182]: should install be commented out?

Comment (johnmcollier) [reply Makefile:182]: Yes, for now. The APIResources and APIExports manage the installation of the CRDs.

File: config/kcp/kustomization.yaml [hunk 3]
```diff
@@ -6,39 +6,15 @@ namespace: application-service-system
...
-# HAS on KCP needs to use default service account name right now
-- patch: |-
-    - op: replace
-      path: /spec/template/spec/serviceAccountName
-      value: default
-  target:
-    kind: Deployment
-    name: controller-manager
\ No newline at end of file
+- apiresourceschema_has.yaml
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply config/kcp/kustomization.yaml:20]: No. That needs to be applied separately, as the APIExport needs to have been created first.

**Pull request merged**